src/components/core/geoChart/GeoChartLegendRenderer.tsx

```diff
@@ -1,70 +1,107 @@
+import { IGeoData, IPushpinCategoryLegendItem } from "../../../interfaces/GeoChart";
+import { getAvailableLegends } from "../../../helpers/geoChart/data";
 
 export interface IGeoChartLegendRendererProps {
-    config: IGeoConfig;
-    geoData: IGeoData;
+    categoryItems?: IPushpinCategoryLegendItem[]; // used for Category legend
+    format?: string;
+    geoData?: IGeoData; // used for Color/Size legend
+    height?: number;
+    locale?: string;
     position?: string;
-    categoryItems?: IPushpinCategoryLegendItem[];
-    onItemClick?(item: IPushpinCategoryLegendItem): void;
+    responsive?: boolean;
+    showFluidLegend?: boolean;
+    onItemClick?: (item: IPushpinCategoryLegendItem) => void;
 }
 
 export default function GeoChartLegendRenderer(props: IGeoChartLegendRendererProps): JSX.Element {
-    const numericSymbols = useNumbericSymbols();
-    const { position = TOP, categoryItems, geoData } = props;
-    const { size, color } = geoData;
-
-    const hasSizeData = Boolean(size && size.data.length);
-    const hasSegmentData = Boolean(categoryItems && categoryItems.length);
-    const isColorLegendVisible = Boolean(color && color.data.length && !hasSegmentData);
-    const isLegendVisible = isColorLegendVisible || hasSizeData || hasSegmentData;
+    const { categoryItems, geoData, position, responsive, showFluidLegend } = props;
+    const { hasCategoryLegend, hasColorLegend, hasSizeLegend } = getAvailableLegends(categoryItems, geoData);
 
+    const isLegendVisible = hasCategoryLegend || hasColorLegend || hasSizeLegend;
     if (!isLegendVisible) {
         return null;
     }
 
-    const classes = cx("geo-legend s-geo-legend", `position-${position}`, { "has-size-legend": hasSizeData });
+    const isFluidLegendEnabled = responsive && showFluidLegend;
+    const isBottomPosition = isFluidLegendEnabled || position === BOTTOM;
+
+    const classNames = cx("geo-legend", "s-geo-legend", {
+        "viz-fluid-legend-wrap": isFluidLegendEnabled,
+        "viz-static-legend-wrap": !isFluidLegendEnabled,
+        "has-size-legend": hasSizeLegend,
+        [`position-${position}`]: !isBottomPosition,
+        "position-bottom": isBottomPosition,
+    });
 
     return (
-        <div className={classes}>
-            {isColorLegendVisible && renderPushpinColorLegend(color, numericSymbols)}
-            {hasSegmentData && renderPushpinCategoryLegend(props)}
-            {hasSizeData && renderPushpinSizeLegend(size, numericSymbols)}
-        </div>
+        <Measure client={true}>
+            {({ measureRef, contentRect }: MeasuredComponentProps) => {
+                return (
+                    <div className={classNames} ref={measureRef}>
+                        {hasColorLegend && renderPushpinColorLegend(props)}
+                        {hasCategoryLegend && renderPushpinCategoryLegend(props, contentRect, hasSizeLegend)}
+                        {hasSizeLegend && renderPushpinSizeLegend(props)}
```

Comment:
Please move condition into separate function
```suggestion
                        {renderPushpinColorLegend(props)}
                        {renderPushpinCategoryLegend(props, contentRect, hasSizeLegend)}
                        {renderPushpinSizeLegend(props)}
 ..................
 renderPushpinColorLegend = (...) => {
      if (! hasColorLegend) { return null; }
 }  
```

Reply:
to fix your comment, I have to code like this 
```
renderPushpinColorLegend(props, hasColorLegend)
...
renderPushpinColorLegend = (props, hasColorLegend) => {
      if (!hasColorLegend) { return null; }
 }  
```
Is that a good idea?

Reply:
Just minor comment, it's up to you.
That's mean
```
renderPushpinColorLegend = (props, hasColorLegend) => {
      if (!hasColorLegend) { return null; }
      const {
            geoData: {
                color: { data, format },
            },
            position,
            responsive,
            showFluidLegend,
        } = props;
      .....................
 } 
```

Reply:
updated